contract/src/lib.rs

```diff
@@ -360,91 +399,223 @@ impl Contract {
+            reference_hash: None,
         }
     }
 
+    fn answers() -> Vec<Answer> {
+        vec![
+            Answer {
+                id: 1,
+                article_id: 2,
+                your_answer: "yes".to_string(),
+                correct_answer: "no".to_string(),
+                pass: true,
+            },
+            Answer {
+                id: 2,
+                article_id: 1,
+                your_answer: "yes".to_string(),
+                correct_answer: "yes".to_string(),
+                pass: true,
+            },
+            Answer {
+                id: 3,
+                article_id: 1,
+                your_answer: "yes".to_string(),
+                correct_answer: "no".to_string(),
+                pass: true,
+            },
+        ]
+    }
+
+    fn get_context(signer: &AccountId, deposit: Option<u128>) -> VMContextBuilder {
+        let mut contex = VMContextBuilder::new();
+        contex
+            .current_account_id(account_id())
+            .account_balance(100 * ONE_NEAR)
+            .signer_account_id(signer.clone())
+            .attached_deposit(deposit.unwrap_or(0))
+            .prepaid_gas(Gas(30_000_000_000_000))
+            .is_view(false);
+        contex
+    }
+
+    //check status of contract initiation after initiation, expect true
+    #[test]
+    fn get_status_init() {
+        let context = get_context(&owner_id(), None);
+        testing_env!(context.build());
+        let contract = Contract::new_default_meta(owner_id());
+        let status = contract.get_status_init();
+        assert_eq!(status, true)
+    }
+
+    //set subjects exam, get subject by name 'chemistry' and expect that result.length == chemistry.length
     #[test]
-    fn set_then_get_greeting() {
-        let context = get_context(vec![], false);
-        testing_env!(context);
-        let mut contract = Welcome::default();
-        contract.set_greeting("howdy".to_string());
+    fn set_subjects() {
+        let context = get_context(&owner_id(), None);
+        testing_env!(context.build());
+        let mut contract = Contract::new_default_meta(owner_id());
+        let chemistry = chemistry();
+        contract.set_subjects();
+
         assert_eq!(
-            "howdy".to_string(),
-            contract.get_greeting("bob_near".to_string())
+            contract.get_tickets_by_subject_name(&String::from("chemistry")).len(),
+            chemistry.len()
         );
     }
 
+    //set tickets for chemistry exam and expect that result.length == chemistry.length
+    #[test]
+    fn set_tickets() {
+        let context = get_context(&owner_id(), None);
+        testing_env!(context.build());
+        let mut contract = Contract::new_default_meta(owner_id());
+        let chemistry = chemistry();
+        contract.set_tickets(String::from("chemistry"), chemistry.clone());
+        let result = contract.get_tickets_by_subject_name(&String::from("chemistry"));
+
+        assert_eq!(result.len(),
```

Comment:
Strange formatting. Are you using rustfmt tool? It shouldn't format like this.

Reply:
Got it, thanks.

Reply:
Got it, thanks.